contracts/FlowTransactionScheduler.cdc

```diff
@@ -0,0 +1,1440 @@
+            data: data, 
+            timestamp: timestamp, 
+            priority: priority, 
+            executionEffort: executionEffort, 
+            fees: <-fees
+        )
+    }
+
+    access(all) fun estimate(
+        data: AnyStruct?,
+        timestamp: UFix64,
+        priority: Priority,
+        executionEffort: UInt64
+    ): EstimatedScheduledTransaction {
+        return self.sharedScheduler.borrow()!
+            .estimate(
+                data: data, 
+                timestamp: timestamp, 
+                priority: priority, 
+                executionEffort: executionEffort,
+            )
+    }
+
+    access(all) fun cancel(scheduledTx: @ScheduledTransaction): @FlowToken.Vault {
+        let id = scheduledTx.id
+        destroy scheduledTx
+        return <-self.sharedScheduler.borrow()!.cancel(id: id)
+    }
+
+    /// getTransactionData returns the transaction data for a given ID
+    /// This function can only get the data for a transaction that is currently scheduled or pending execution
+    /// because finalized transaction metadata is not stored in the contract
+    /// @param id: The ID of the transaction to get the data for
+    /// @return: The transaction data for the given ID
+    access(all) view fun getTransactionData(id: UInt64): TransactionData? {
+        return self.sharedScheduler.borrow()!.getTransaction(id: id)
+    }
+
+    access(all) view fun getCanceledTransactions(): [UInt64] {
+        return self.sharedScheduler.borrow()!.getCanceledTransactions()
+    }
+
+    access(all) view fun getStatus(id: UInt64): Status? {
+        return self.sharedScheduler.borrow()!.getStatus(id: id)
+    }
+
+    /// getTransactionsForTimeframe returns the IDs of the transactions that are scheduled for a given timeframe
+    /// @param startTimestamp: The start timestamp to get the IDs for
+    /// @param endTimestamp: The end timestamp to get the IDs for
+    /// @return: The IDs of the transactions that are scheduled for the given timeframe
+    access(all) fun getTransactionsForTimeframe(startTimestamp: UFix64, endTimestamp: UFix64): {UFix64: {UInt8: [UInt64]}} {
+        return self.sharedScheduler.borrow()!.getTransactionsForTimeframe(startTimestamp: startTimestamp, endTimestamp: endTimestamp)
+    }
+
+    access(all) view fun getSlotAvailableEffort(timestamp: UFix64, priority: Priority): UInt64 {
+        return self.sharedScheduler.borrow()!.getSlotAvailableEffort(timestamp: timestamp, priority: priority)
+    }
+
+    access(all) fun getSchedulerConfigurationDetails(): {SchedulerConfig} {
+        return self.sharedScheduler.borrow()!.getConfigurationDetails()
+    }
+    
+    /// getSizeOfData takes a transaction's data
+    /// argument and stores it in the contract account's storage, 
+    /// checking storage used before and after to see how large the data is in MB
+    /// If data is nil, the function returns 0.0
+    access(all) fun getSizeOfData(_ data: AnyStruct?): UFix64 {
```

Comment:
This function can dissable operation of scheduling if the account balance gets too low. We should probably add a comment at the top of the contract that the account holding the resource should have at least X Flow (calculate X) to operate.

Reply:
Why would it disable scheduling? It saves and loads, so after the function is done executing, the storage used is not any different. plus, any fees that are used in scheduling are deposited to the contract account's vault, so that would cover any storage fees anyway